packages/storefront-lwc/scripts/extension/productDetailExtension.mjs

```diff
@@ -0,0 +1,103 @@
+
+import {core, API_EXTENSIONS_KEY} from '@sfcc-core/core';
+import { resolverFactory } from "@sfcc-core/core-graphql";
+import apolloServerCore from 'apollo-server-core';
+import CommerceSdk from 'commerce-sdk';
+import Image from '../../../@sfcc-bff/productapi/src/api/models/Image';
+
+const { gql } = apolloServerCore;
+
+// Demo : Define the Product Recommendation extension to the Product object
+const productRecommendationTypeDef = gql`
+    type Recommendation {
+        productId: String
+        productName: String
+        image: Image
+    }
+    extend type Product {
+        recommendations: [Recommendation]
+    }
+`;
+
+// Resolve the product recommendations for a Product
+const productRecommendationResolver = (config) => {
+    let product_recommendations = [];
+    return {
+        Product: {
+            recommendations: async (product) => {
+                if (product.recommendations) {
+                    let productIds  = [];
+                    let ids
+                    product.recommendations.forEach(recommendation => {
+                        productIds.push(recommendation.recommendedItemId);
+                        ids = productIds.toString();
+                    });
+                    const result = await getClientProducts(config, ids);
+                    result.data.forEach(apiProduct => {
+                        console.log('apiProduct is ', apiProduct);
+                        product_recommendations.push({
+                            productId: apiProduct.id, 
+                            productName:apiProduct.name, 
+                            image: new Image(apiProduct.imageGroups[2].images[0])
+                        })
+                    })
+                    return product_recommendations;
+                }
+            }
+        }
+    }
+}
+
+const getClientProducts = async (config, ids) => {
+    const clientId = config.COMMERCE_CLIENT_CLIENT_ID;
```

Comment:
should we be using the same pattern as Ann's current/recent PR with new SDK where we have a utility

Reply:
sure, I updated per your request! Thank you for your feedback!